src/backend/joanie/core/models/certifications.py

```diff
@@ -87,9 +91,17 @@ def document(self):
         Get the document related to the certificate instance.
         """
         document_issuer = import_string(self.certificate_definition.template)
-        context = self.get_document_context()
-        document = document_issuer(identifier=self.id, context_query=context)
-        return document.create(persist=False)
+        try:
+            context = self.get_document_context()
+        except ValueError as exception:
+            logger.error(
+                "Cannot get document context to generate certificate.",
+                exc_info=exception,
+            )
+            return None
+        else:
```

Comment:
As the "except" block has a return, you can remove this "else" block :)

Reply:
Oh yes. Good catch